pkg/reconciler/pipelinerun/pipelinerun_test.go

```diff
@@ -507,11 +413,9 @@ spec:
 	tr1Name := "test-pipeline-run-success-unit-test-1"
 	tr2Name := "test-pipeline-run-success-unit-test-cluster-task"
 
-	verifyTaskRunStatusesCount(t, reconciledRun.Status, 2)
+	// TODO
+	// verifyTaskRunStatusesCount(t, reconciledRun.Status, 2)
```

Comment:
I was trying to verify if this shall be changed to 3

Reply:
It doesn't look like you've changed the number of tasks in the input data (but I could be wrong) so I'd be surprised if this is changed. Does the output of the test align with your expectations, given the input of the test?

Reply:
I have been trying to understand https://github.com/tektoncd/pipeline/blob/2da10fc5ff9d99571b6055bcb86aca12805cb9e0/pkg/reconciler/pipelinerun/pipelinerun_test.go#L161 about how the taskruns are fetched from actions, but not sure how to investigate in the ktesting.CreateAction here

Reply:
I'm confused; is the problem with `getTaskRunCreations` on line 375, or with `verifyTaskRunStatusesCount` on line 417?

It looks like `verifyTaskRunStatusesCount` just counts the number of TaskRuns in the PipelineRun.Status.ChildReferences, which shouldn't have changed.

If the problem is with `getTaskRunCreations`, that's a bit more complicated. I don't like the usage of that function in our test suite, as it verifies implementation details (API calls) rather than the outputs we care about (TaskRuns themselves), making the tests harder to understand. We should refactor our tests to not rely on this, but that's out of scope for this PR.

If you want to better understand the Action class, you could try stepping through in a debugger (or use print statements) to figure out what api calls are being made (aka the actions), and if anything has changed that you wouldn't expect to change? Unfortunately, the Action class is undocumented, as is the client-go testing package (https://pkg.go.dev/k8s.io/client-go/testing#Action), but perhaps you could look for example usages of this package to better understand what the actions are.

Reply:
Thanks the clarifications Lee. Yes the issue is with `verifyTaskRunStatusesCount `, and it changed to 3 after I deleted some of the resources (which feels wrong). And that's when I looked into how the taskrun status are being created and led me to the confusion at `getTaskRunCreations`.

And appreciate it for the pointers to the doc

Reply:
Can you print the output pipelinerun and make sure it has only the taskruns that are expected? This might be easier than trying to dig into `getTaskRunCreations`.

Reply:
Thanks Lee! I figured this out earlier following your suggestions above. 

Sorry haven't been updating this PR yet. I will check again before bring this ready for review, and create an followup issue on the cleanups for `getTaskRunCreations`.

Reply:
Openend https://github.com/tektoncd/pipeline/issues/6143 and this shall be ready to review:) Thanks!